lib/model/promotion.rb

```diff
@@ -24,6 +24,11 @@ def apply(line_items)
       eligible = sorted.filter { |item| @eligible_skus.include? item.sku }
       return if eligible.empty?
 
+      # PRO: Great catch here on the edge chase, since we should allow
+      # getting a discount on your second COCOA box, but not if you're buying a singele one.
+      #
+      # In this case since you're working with line items object, the comparison here will work
+      # as they're not the same object, but the same line item.
```

Comment:
Yup, the default object equality operator here works great, since I'm just checking if "is this trying to discount itself?"